packages/synapse-react-client/src/types/tanstack-table-augmentation.d.ts

```diff
@@ -12,6 +12,7 @@ type ColumnFilterEnumValue<TValue = unknown> = {
 declare module '@tanstack/react-table' {
   interface TableMeta<TData extends RowData> {
     rowSet?: RowSet
+    hideAccessIconForExternalFileHandle?: boolean
```

Comment:
Instead of here, can you put this property in `SynapseTableContext`, which is specific to `SynapseTable`? This `TableMeta` would ideally be general to other tables in the UI (we have clearly not been great at this, so putting it here was a reasonable choice)